src/plugins/index_pattern_field_editor/public/components/field_editor/form_schema.ts

```diff
@@ -7,12 +7,72 @@
  */
 
 import { i18n } from '@kbn/i18n';
-import { fieldValidators } from '../../shared_imports';
+import { EuiComboBoxOptionOption } from '@elastic/eui';
+import type { Subscription } from 'rxjs';
+import { first } from 'rxjs/operators';
+import { PainlessLang } from '@kbn/monaco';
 
+import { fieldValidators, FieldConfig, RuntimeType, ValidationFunc } from '../../shared_imports';
+import type { Context } from '../preview';
 import { RUNTIME_FIELD_OPTIONS } from './constants';
 
+type CancelablePromise = Promise<any> & { cancel?(): void };
```

Comment:
Actually it can't be `void` as we need to be able to return a validation error. It made me realised that the form lib could export the cancelable promise with the correct Promise type. (https://github.com/elastic/kibana/pull/109233/commits/d25bf922273e706b55ed6b6451a2602ff0a43b90)